File: app/src/main/java/com/designpatterns/demo/behavioral/template/java/MoneyMarketAccount.java
package com.designpatterns.demo.behavioral.template.java;

/**
 * 具体模板角色类
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/6/7 上午11:40
 * @project_name DesignPatternsDemo
 */
public class MoneyMarketAccount extends Account {
    @Override
    protected String doCalculateAccountType() {
        return "Money Market";
    }

    @Override
    protected double doCalculateInterestRate() {
        return 0.045;
    }
}

File: app/src/main/java/com/designpatterns/demo/behavioral/template/kotlin/MoneyMarketAccount.kt
package com.designpatterns.demo.behavioral.template.kotlin

/**
 * 具体模板角色类
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/6/7 下午1:40
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class MoneyMarketAccount:Account(){


    override fun doCalculateAccountType(): String ="Money Market"

    override fun doCalculateInterestRate(): Double=0.045

}